scalableminds/webknossos
# Support uint64 segmentations (up to 2^53 - 1)

- note that the backend will support only segment ids up to (2^63)-1 (scala/java long type)
- note that the frontend will only support segment ids up to (2^53)-1 (javascript number type)
- the back-end asserts the smaller (2^53)-1 constraint
- JS-side mappings (aka JSON mappings) are only supported for 32 bits right now. If we want to improve this, cuckoo hashing could become useful again, but I'm not sure about the priority here which is why I'd defer this.

### Todo:
- [X] warn user about merger mode support being limited to 32-bits
- [x] warn user about mapping support being limited to 32-bits

### URL of deployed dev instance (used for testing):
- https://uint.webknossos.xyz/datasets/sample_organization/64bit_segmentation/view#12,16,0,0,0.164

### Steps to test:
- I created a test dataset and uploaded it to the dev instances (see [here](https://www.notion.so/scalableminds/Test-Data-c0563be9c4a4499dae4e16d9b2497cfb#0412f70b2dfc4b1988e3ef4cea05c738) for details; including how this was produced to see where to expect which voxels)
- test normal viewing of the dataset
  - differing segment ids should have different colors and should be isolated on hover
- create an annotation with the existing layer
  - test brushing
  - test both overwrite modes
  - test the segment list view
  - test isosurface rendering
  - test the merger mode (it should warn that only 32-bit values are valid)
  - activate a json mapping (it should warn that only 32-bit values are valid)
  - test the volume interpolation feature
  - test the flood fill

### Issues:
- fixes #4852 

------
(Please delete unneeded items, merge only when none are left open)
